common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt

```diff
@@ -184,6 +200,35 @@ class NetworkSensorManager : SensorManager {
     private fun hasWifi(context: Context): Boolean =
         context.applicationContext.getSystemService<WifiManager>() != null
 
+    @SuppressLint("PrivateApi")
+    private fun hasHotspot(context: Context): Boolean {
+        val wifiManager: WifiManager = context.applicationContext.getSystemService()!!
+        return try {
+            wifiManager.javaClass.getDeclaredMethod("isWifiApEnabled")
+            true
+        } catch (e: NoSuchMethodException) {
+            false
+        }
+    }
+    private fun updateHotspotEnabledSensor(context: Context) {
+        if (!isEnabled(context, hotspotState) || !hasWifi(context) || !hasHotspot(context)) {
```

Comment:
This should be an `and` condition as all 3 need to be met in order to be used

Reply:
How should this be? I think this is the correct method, because if any of those are `false`, it should return.

Reply:
Oh sorry you are correct. In fact we dont need to add `hasWifi` and `hasHotspot` methods in this check as its already done up above in `getAvailableSensors`